src/main/java/net/dries007/tfc/common/blocks/crop/DeadDoubleCropBlock.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.blocks.crop;
+
+
+import java.util.List;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.EnumProperty;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.common.blocks.TFCBlockStateProperties;
+
+public class DeadDoubleCropBlock extends DeadCropBlock
+{
+    public static final EnumProperty<DoubleCropBlock.Part> PART = TFCBlockStateProperties.DOUBLE_CROP_PART;
+
+    public DeadDoubleCropBlock(Properties properties, Crop crop)
+    {
+        super(properties, crop);
+        registerDefaultState(getStateDefinition().any().setValue(PART, DoubleCropBlock.Part.BOTTOM).setValue(MATURE, false));
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder)
+    {
+        super.createBlockStateDefinition(builder.add(PART));
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context)
+    {
+        return state.getValue(PART) == DoubleCropBlock.Part.BOTTOM ? CropBlock.FULL_SHAPE : CropBlock.HALF_SHAPE;
+    }
+
+    @Override
+    public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos)
+    {
+        final DoubleCropBlock.Part part = state.getValue(PART);
+        final BlockState belowState = level.getBlockState(pos.below());
+        if (part == DoubleCropBlock.Part.BOTTOM)
+        {
+            return TFCTags.Blocks.FARMLAND.contains(belowState.getBlock());
+        }
+        else
+        {
+            return belowState.is(this) && belowState.getValue(PART) == DoubleCropBlock.Part.BOTTOM;
+        }
+    }
+
+    @Override
+    public void addHoeOverlayInfo(Level level, BlockPos pos, BlockState state, List<Component> text, boolean isDebug)
+    {
+        super.addHoeOverlayInfo(level, state.getValue(PART) == DoubleCropBlock.Part.TOP ? pos.below() : pos, state, text, isDebug);
```

Comment:
This technically breaks the contract of the given method, it should be passing in the value of `level.getBlockState(pos.below())` if it changes the position.